### PR TITLE
Week 2 homework Sila Kaymak

### PR DESCRIPTION
**Pull request by bsilakaymak**

### CODE REVIEW

File: Week2/homework/grade.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        console.log(" You got an " + letterGrade +" ("+ score + "%)");
+    }
+}
+
+calcGrade(calcPerc(85, 100));
+// other examples
+calcGrade(calcPerc(27, 50));
+calcGrade(calcPerc(65, 120));
+calcGrade(calcPerc(7.5, 12));
+
```
Comment (nudge) [inline Week2/homework/grade.js]: :+1: Very nice!

Comment (nudge) [inline Week2/homework/grade.js:43]: :+1: Very nice!

File: Week2/homework/js-exercises/1-correct-the-sentence.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+"use strict";
+let myString = 'hello,this,is,a,difficult,to,read,sentence';
+
+console.log(myString.length);
+
+let myNewString = myString.replace(/,/g, ' ');
+
+console.log(myNewString);
```
Comment (nudge) [inline Week2/homework/js-exercises/1-correct-the-sentence.js:8]: :+1:

You can also acheive the same without using regular expressions by using `.split()` and `.join()` methods on `String`. You can look up MDN on how they work :)

File: Week2/homework/js-exercises/2-even-or-odd.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+"use strict";
+
+for(let i = 0; i <= 20; i++){
+   if (i % 2 === 0){
+       console.log("The number " +i+ " is even!")
+   }else{
+    console.log("The number " +i+ " is odd!")  
+   }
+}
```
Comment (nudge) [inline Week2/homework/js-exercises/2-even-or-odd.js:9]: 👍

File: Week2/homework/js-exercises/3-sarma-recipe.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+const myRecipe = {
+    name: "sarma",
+    ingredients: ["rice", "grape leaves", "tomato pasta", "pomegranade syrup", "raisins", "nuts"],
+    servings :"4",
+};
+const myRecipeKeys = Object.keys(myRecipe);
+for (let i = 0; i < myRecipeKeys.length; i++) {
+    console.log(myRecipeKeys[i] + ": " + myRecipe[myRecipeKeys[i]])  
+};
```
Comment (nudge) [inline Week2/homework/js-exercises/3-sarma-recipe.js:11]: 👍

File: Week2/homework/js-exercises/4-reading-list.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const bookKeys = Object.keys(bookList[i]);
+for(i = 0; i < bookList.length; i++){
+    console.log(bookList[i]);
+}
+
+console.log("<---->")
+
+//Title and author
+for(i = 0; i < bookList.length; i++){
+console.log(bookList[i][bookKeys[0]] + " by " + bookList[i][bookKeys[1]]);
```
Comment (nudge) [inline Week2/homework/js-exercises/4-reading-list.js]: Although this works, it is much easier to read code that looks like this:
```
for(i = 0; i < bookList.length; i++) {
  console.log(bookList[i].title + " by " + bookList[i].author);
}
```

In addition, if you decide to add more keys to the `bookList` object, the ordering of keys might change, so it is not guaranteed that 0 will be title, 1 will be author, etc.

File: Week2/homework/js-exercises/4-reading-list.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+// Two logs showing already read or not
+
+for(i = 0; i < bookList.length; i++){
+    if (bookList[i][bookKeys[2]] === true){
+        console.log("You already read " +"\"" + bookList[i][bookKeys[0]]+"\"" )
+    } else {
+        console.log("You still need to read " + "\"" + bookList[i][bookKeys[0]]  + "\"")
+
+    }
+}
```
Comment (nudge) [inline Week2/homework/js-exercises/4-reading-list.js:50]: 👍 
Same comment as above -- it is preferred to use `.` syntax to access object keys as it makes it much easier to read.

File: Week2/homework/js-exercises/5-drinks.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+/* I know this is not the way I was supposed to follow. It gives the desired result but I should have created a conditional and then
+push.I did a lot of research on it, I tried to use .filter to see the number of occurences but I didn't get it fully so
+I decided not to use it until I learn it better.*/
+
+for(let i = 0; i < 5; i++){
+   drinkTray.push(drinkTypes[i % 3]) 
+}
+console.log(drinkTray);
+console.log("Hey guys I brought a " + drinkTray + "!");
```
Comment (nudge) [inline Week2/homework/js-exercises/5-drinks.js:16]: 👍 
True that using filter makes this exercise much easier. You will learn about this in JS2.

Another solution might be:
```
const drinkTypes = ["cola", "lemonade", "water"];
let drinkTray = [];
let currentDrinkIndex = 0;
do {
  let currentDrinkType = drinkTypes[currentDrinkIndex];
  let currentDrinkCounter = 0;
  for (let i=0 ; i<drinkTray.length ; i++) {
    if (drinkTray[i] === currentDrinkType) {
      currentDrinkCounter++;
    }
  }
  if (currentDrinkCounter < 2) {
    drinkTray.push(currentDrinkType);
  } else {
    currentDrinkIndex++;
  }
} while (drinkTray.length < 5)
console.log(drinkTray);
console.log("Hey guys I brought a " + drinkTray + "!");
```

**Comment by bsilakaymak** [comment]
Thanks a lot for your time:) I changed the one with object keys to dot syntax, I guess it is much safer to use it. I will also check other possible solutions you mentioned for drinks exercise and the one I used a regular expression to remove the comma.

**Comment by wouterkleijn** [comment]
Thanks a lot @bsilakaymak for addressing @nudge 's comments. Well done!

**Pull request closed without merging**